src/modules/controllers/task.controller.js

```diff
@@ -0,0 +1,74 @@
+const Task = require('../../models/task');
+
+const getAllTasks = async (req, res) => {
+  try {
+    const result = await Task.find();
+    res.status(200).send({ data: result })
+  } catch (error) {
+    res.status(400).send('Bad Request tasks not found')
+    console.error('tasks not found');
+  }
+}
+
+const createNewTask = async (req, res) => {
+  try {
+    const task = new Task(req.body);
+    const result = await task.save();
+    res.status(200).send(result);
+  } catch (error) {
+    res.status(400).send('Bad Request task not saved')
+    console.error('task not saved');
+  }
+};
+
+const changeTaskInfo = async (req, res) => {
+  try {
+    const id = req.body._id;
+    const result = await Task.findOneAndUpdate({ _id: id }, req.body)
+    res.status(200).send(result);
+  } catch (error) {
+    res.status(400).send('Bad Request task not changed')
+    console.error('task not changed');
+  }
+};
+
+const changeTaskComplete = async (req, res) => {
+  try {
+    const id = req.body._id;
+    const result = await Task.findOneAndUpdate({ _id: id }, req.body);
```

Comment:
You in this request and in request above update all object at once. But you need either only text, or only a checkbox. Otherwise, what is the difference between these queries? They are the same after all